pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/model/event/CreateKitEventAction.java

```diff
@@ -0,0 +1,129 @@
+import org.broadinstitute.ddp.db.dto.kit.KitConfigurationDto;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.model.address.MailAddress;
+import org.broadinstitute.ddp.model.kit.KitConfiguration;
+import org.broadinstitute.ddp.model.user.User;
+import org.broadinstitute.ddp.monitoring.PointsReducerFactory;
+import org.broadinstitute.ddp.monitoring.StackdriverCustomMetric;
+import org.broadinstitute.ddp.monitoring.StackdriverMetricsTracker;
+import org.broadinstitute.ddp.pex.PexInterpreter;
+import org.broadinstitute.ddp.service.DsmAddressValidationStatus;
+import org.broadinstitute.ddp.service.KitCheckService;
+import org.jdbi.v3.core.Handle;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.broadinstitute.ddp.service.DsmAddressValidationStatus.DSM_INVALID_ADDRESS_STATUS;
+
+@Slf4j
+public class CreateKitEventAction extends EventAction {
+
+    Long kitTypeId;
+    /**
+     * Key is study guid, value is the metrics transmitter for the study
+     */
+    private static final Map<String, StackdriverMetricsTracker> kitCounterMonitorByStudy = new HashMap<>();
+
+    public CreateKitEventAction(EventConfiguration eventConfiguration, EventConfigurationDto dto) {
+        super(eventConfiguration, dto);
+        kitTypeId = dto.getKitTypeId();
+    }
+
+    @Override
+    public void doAction(PexInterpreter pexInterpreter, Handle handle, EventSignal signal) {
+        Integer delayBeforePosting = eventConfiguration.getPostDelaySeconds();
+        if (delayBeforePosting != null && delayBeforePosting > 0) {
+            if (!eventConfiguration.dispatchToHousekeeping()) {
+                throw new DDPException("Incompatible event configuration:"
+                        + " delayed Kit Creation events should set dispatchToHousekeeping");
+            }
+            long queuedEventId = queueDelayedEvent(handle, signal);
+            log.info("Queued Create Kit event with id {}", queuedEventId);
+        } else {
+            doActionSynchronously(handle, signal);
+        }
+    }
+
+    public void doActionSynchronously(Handle handle, EventSignal signal) {
+
+        List<KitConfigurationDto> kitConfigs = handle.attach(KitConfigurationDao.class)
+                .getKitConfigurationDtosByStudyId(signal.getStudyId());
+
+        //load user and address
+        DsmAddressValidationStatus statusType = null;
+        User user = handle.attach(UserDao.class).findUserByGuid(signal.getParticipantGuid()).get();
+        MailAddress defaultAddress = handle.attach(JdbiMailAddress.class)
+                .findDefaultAddressForParticipant(signal.getParticipantGuid())
+                .orElse(null);
+
+        if (defaultAddress == null) {
+            log.warn("Participant {} is missing a default mailing address. Deleting the create kit event", signal.getParticipantGuid());
```

Comment:
Do we have a way for admins or operators (i.e.. us) to become aware that this situation has arisen? If we have to depend on logs, then perhaps this ought to be an error?